File: iOS/Objective C/HelloViews/HelloViews/TapViewController.m
//
//  TapViewController.m
//  HelloViews
//
//  Created by Amirali Rajan on 10/2/12.
//  Copyright (c) 2012 __MyCompanyName__. All rights reserved.
//

#import "TapViewController.h"

@interface TapViewController ()

@end

@implementation TapViewController

- (id)initWithNibName:(NSString *)nibNameOrNil bundle:(NSBundle *)nibBundleOrNil
{
    self = [super initWithNibName:nibNameOrNil bundle:nibBundleOrNil];
    if (self) {
        // Custom initialization
    }
    return self;
}

- (void)viewDidLoad
{
    [super viewDidLoad];
    // Do any additional setup after loading the view from its nib.
    
    self.title = [NSString stringWithFormat: @"Tap (%d)", self.navigationController.viewControllers.count];
    
    UIBarButtonItem *rightButton = [[UIBarButtonItem alloc] initWithTitle:@"Push" style:UIBarButtonItemStyleBordered target:self action:@selector(rightButtonClicked:)];
    self.navigationItem.rightBarButtonItem = rightButton;
}

-(id) init
{
    self = [super init];
    
    self.tabBarItem = [[UITabBarItem alloc] initWithTabBarSystemItem:UITabBarSystemItemFavorites tag:1];
    
    return self;
}
                                   
-(void)rightButtonClicked:(id)sender
{
    TapViewController *controller = [[TapViewController alloc] init];
    
    [self.navigationController pushViewController:controller animated:true];
                                   
}

- (void)viewDidUnload
{
    [super viewDidUnload];
    // Release any retained subviews of the main view.
    // e.g. self.myOutlet = nil;
}

- (BOOL)shouldAutorotateToInterfaceOrientation:(UIInterfaceOrientation)interfaceOrientation
{
    return (interfaceOrientation == UIInterfaceOrientationPortrait);
}

@end

File: iOS/monotouch/HelloViews/HelloViews/HelloViews/TapViewController.cs

using System;
using System.Drawing;

using MonoTouch.Foundation;
using MonoTouch.UIKit;

namespace HelloViews
{
	public partial class TapViewController : UIViewController
	{
		public TapViewController () : base ("TapViewController", null)
		{
			this.TabBarItem = new UITabBarItem(UITabBarSystemItem.Favorites, 1);
		}
		
		public override void DidReceiveMemoryWarning ()
		{
			// Releases the view if it doesn't have a superview.
			base.DidReceiveMemoryWarning ();
			
			// Release any cached data, images, etc that aren't in use.
		}
		
		public override void ViewDidLoad ()
		{
			base.ViewDidLoad ();
			
			// Perform any additional setup after loading the view, typically from a nib.
			Title = string.Format("Tap ({0})", this.NavigationController.ViewControllers.Length);

			var rightButton = new UIBarButtonItem("Push",UIBarButtonItemStyle.Bordered, this, new MonoTouch.ObjCRuntime.Selector("push"));
			this.NavigationItem.RightBarButtonItem = rightButton;
		}

		[Export("push")]
		public void push ()
		{
			var new_controller = new TapViewController();
			this.NavigationController.PushViewController(new_controller, true);
		}

		[Obsolete]
		public override void ViewDidUnload ()
		{
			base.ViewDidUnload ();
			
			// Clear any references to subviews of the main view in order to
			// allow the Garbage Collector to collect them sooner.
			//
			// e.g. myOutlet.Dispose (); myOutlet = null;
			
			ReleaseDesignerOutlets ();
		}

		[Obsolete]
		public override bool ShouldAutorotateToInterfaceOrientation (UIInterfaceOrientation toInterfaceOrientation)
		{
			// Return true for supported orientations
			return (toInterfaceOrientation != UIInterfaceOrientation.PortraitUpsideDown);
		}
	}
}

